tests/unit/pudl_archiver_test.py

```diff
@@ -50,7 +49,7 @@ def failed_run():
 @pytest.mark.asyncio
 async def test_archive_datasets(successful_run, failed_run, unchanged_run, mocker):
     """Test that archive datasets creates run_summary file."""
-    open_mock = mocker.patch("pudl_archiver.open")
+    open_mock = mocker.patch("pudl_archiver.Path.open")
```

Comment:
This is the mocking that I think is related to the one unit test failure.

Reply:
The problem here is that `open` is a function that returns file contents, and `Path.open` is an instance method - so you need to create a whole `Path` object and then mock _its_ `.open()` method.

Reply:
I was wrong about this - the mocking is actually working fine as is, but since we pass in a string (`"file"`) to `yaml.safe_load()` it returns a string.

for example:

```python
>>> import yaml
>>> yaml.safe_load("file")
'file'
>>> yaml.safe_load("- file")
['file']
>>> yaml.safe_load("file: a")
{'file': 'a'}
```

So we need to give it better fake data. I'm not sure why it worked before...

Reply:
Is it really supposed to be affecting the `yaml.safe_load()` at all?  It seemed like a far-flung side effect of the mocking -- which maybe makes sense if I was accidentally overwriting the classmethod `Path.open()` rather than a particular instance method.

Reply:
In the old setup, it was just overriding a the `open()` function's behavior in a single module (`pudl_archiver.open`) and not having any effects elsewhere... which seemed like the desired behavior.